include/Factory.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include <map>
+#include "Pieces/Piese.h"
+
+using std::map;
+
+class Piece;
+
+template<typename T>
+class Factory {
+public:
+	using pFnc = unique_ptr<T>(*)();
```

Comment:
`pFnc` isn't a good name for this function pointer type. Try to think of something more readable that will make the reader understand its purpose. Maybe `create_func`?